### PR TITLE
implemented Spring Security and used it

### PR DESCRIPTION
**Pull request by vladkuzya**

### CODE REVIEW

File: src/main/java/mate/academy/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                .httpBasic()
+                .and()
+                .csrf().disable();
+    }
+
+    @Bean
+    public PasswordEncoder getEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
```
Comment (julia-shvabovich) [inline src/main/java/mate/academy/config/SecurityConfig.java:41]: remove this empty line

Comment (julia-shvabovich) [inline src/main/java/mate/academy/config/SecurityConfig.java]: remove this empty line

File: src/main/java/mate/academy/model/dto/UserRegistrationDto.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        fieldMatch = "repeatPassword",
+        message = "Passwords do not match!"
+)
+public class UserRegistrationDto {
+    @EmailValidation
+    private String email;
+    @NotNull
+    @Size(min = 4)
+    private String password;
+    private String repeatPassword;
```
Comment (julia-shvabovich) [inline src/main/java/mate/academy/model/dto/UserRegistrationDto.java:21]: you may also add @NotNull here, but i think it's optional

File: src/main/java/mate/academy/model/dto/request/MovieRequestDto.java [hunk 3]
```diff
@@ -1,6 +1,9 @@
 package mate.academy.model.dto.request;
 
+import javax.validation.constraints.NotNull;
+
 public class MovieRequestDto {
+    @NotNull
```
Comment (julia-shvabovich) [inline src/main/java/mate/academy/model/dto/request/MovieRequestDto.java:6]: the annotation has no sense if you won't add the annotation @Value to your controller methods.
the same is about all of your request dtos, check them all

File: src/main/java/mate/academy/annotation/EmailValidation.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Documented
+@Constraint(validatedBy = EmailValidator.class)
+@Target({ElementType.METHOD, ElementType.FIELD})
```
Comment (RomanDubovskyi) [inline src/main/java/mate/academy/annotation/EmailValidation.java]: do we put this annotation on method?

File: src/main/java/mate/academy/model/dto/request/MovieRequestDto.java (hunk 3 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/model/dto/request/MovieRequestDto.java:6]: @Valid *

File: src/main/java/mate/academy/annotation/EmailValidation.java (hunk 4 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/annotation/EmailValidation.java]: No, of course)

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**